### PR TITLE
docs: replace VuePress default theme with our custom one.

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description
We were having a lot of issues with VuePress default theme styling getting into our dialtone components.

Created a custom theme to avoid such problems and gain even more flexibility to do everything we could need and want.

**Note**
This PR will be the parent of the subsequent ones, this will be merged once all the changes are approved.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

**Review by francisrupert — COMMENTED** [review]
@juliodialpad, this is great stuff. I pulled it down and had fun playing around with `Sidebar.vue` and `SidebarItem.vue` which validated my hopes for being able to completely customize.

Keep at it! Meanwhile, I suppose this should be converted to a draft PR.

**Comment by braddialpad** [comment]
Looking awesome so far, this is going to give us a lot more flexibility. Only thing I'd ask is you maybe give the team a bit of a tutorial when it's ready. This contains a lot of config changes that are a bit hard to follow.

**Comment by juliodialpad** [comment]
> @juliodialpad, this is great stuff. I pulled it down and had fun playing around with `Sidebar.vue` and `SidebarItem.vue` which validated my hopes for being able to completely customize.
> 
> Keep at it! Meanwhile, I suppose this should be converted to a draft PR.

Sure, I'll change this to draft.

I'm keeping this as the "base" PR, this will not be merged until all the styling is finished.

**Comment by juliodialpad** [comment]
> Looking awesome so far, this is going to give us a lot more flexibility. Only thing I'd ask is you maybe give the team a bit of a tutorial when it's ready. This contains a lot of config changes that are a bit hard to follow.

Sure, once this is finished I can give the team a walkthrough on the changes.

**Review by braddialpad — COMMENTED** [review]
Looking pretty fantastic. Couple minor things I noticed before we merge:

Something wrong with spacing... on one page there is huge gaps between paragraphs and on another there is way too little of a gap:


**OLD**
Screenshot 2022-12-06 at 11 40 46 AM
**NEW**
Screenshot 2022-12-06 at 11 40 42 AM

**OLD**
Screenshot 2022-12-06 at 11 39 36 AM
**NEW**
Screenshot 2022-12-06 at 11 39 28 AM

File: docs/about/whats_new/posts/2022-11-18.md [hunk 1]
```diff
@@ -2,9 +2,6 @@
 heading: Release of our new blog!
 author: Brad Paugh
 posted: '2022-11-18'
-prev:
-  text: Back to what's new
-  link: /about/whats_new/
```
Comment (braddialpad) [inline docs/about/whats_new/posts/2022-11-18.md:7]: Because this blog post page is not in the "normal flow" of the sidebar / header it needed this prev link to go back to the what's new page. Any alternative for this with the new system?

Comment (juliodialpad) [reply docs/about/whats_new/posts/2022-11-18.md:7]: I think I can add that link to all blog posts, good catch!

**Review by braddialpad — COMMENTED** [review]
This image seems overly large compared to prod

Screenshot 2022-12-06 at 11 53 27 AM

**Comment by juliodialpad** [comment]
> This image seems overly large compared to prod
> 
> Screenshot 2022-12-06 at 11 53 27 AM

found the issue, fixing

File: docs/about/whats_new/posts/2022-11-18.md (hunk 1 above)
Comment (juliodialpad) [reply docs/about/whats_new/posts/2022-11-18.md:7]: Fixed

**Review by braddialpad — APPROVED** [review]
Looks great :shipit:
